firmware/src/networking/protocol/mod.rs

```diff
@@ -0,0 +1,68 @@
+//! THIS MODULE IS WIP AND NOT FINISHED YET
+//!
+//! Manages the slimevr firmware protocol. All (de)serialization and state associated with
+//! the protocol happens in here.
+#![allow(dead_code)]
+
+mod serialize;
+
+use crate::{Quat, Unreliable};
+
+use embassy_sync::{
+	blocking_mutex::raw::NoopRawMutex,
+	channel::{Channel, Receiver, Sender},
+};
+use firmware_protocol::{CbPacket, SbPacket};
+
+pub struct ImuHandle {
+	signal: &'static Unreliable<Quat>,
+}
+impl ImuHandle {
+	/// Send the latest quaternion. May drop the data.
+	pub fn send_quat(&mut self, q: Quat) {
+		self.signal.signal(q)
+	}
+}
+
+pub struct Protocol {
+	quat_signal: &'static Unreliable<Quat>,
+	sb: Channel<NoopRawMutex, SbPacket, 1>, // TODO: Should these actually be channels?
```

Comment:
These should be Channels; Packets should never be internally dropped.

Reply:
I agree that packets should never be dropped, but we have channels within the same async task, so it might be the wrong concurrent datastructure. Not sure yet.

Reply:
Is one monolithic task better for both networking and protocol? I think they can be separate. Protocol should not be directly linked to what networking impl we use (esp wifi, nrf, ti etc)

Reply:
This struct decouples the logic of the protocol from the transport, so they life in separate files. At that point, whether the stuff is in one async task or split across multiple, im not sure of the right approach.

Reply:
I would say its better to have multiple tasks. Depending on the network impl it might have an impact on receive reliability, and I suspect embassy will be better at efficient scheduling with multiple smaller tasks than few monolithic ones. For example, to send a beacon packet, just awake the network task and do not poll/drive the protocol stack unnecessarily